nemo_curator/modules/semantic_dedup.py

```diff
@@ -200,20 +205,36 @@ def create_embeddings(
 
     def __call__(self, dataset: DocumentDataset) -> DocumentDataset:
         embedding_ddf = self.create_embeddings(dataset.df, self.input_column)
+        t0 = time.time()
         if self.write_embeddings_to_disk:
-            write_to_disk(
-                embedding_ddf,
-                self.embedding_output_dir,
-                write_to_filename=self.write_to_filename,
-                output_type="parquet",
-            )
-            return DocumentDataset(
+            with performance_report_if(
+                self.config.profile_dir,
+                f"embedding-creator-{datetime.now().strftime('%Y%m%d_%H%M%S')}.html",
+            ):
+                write_to_disk(
+                    embedding_ddf,
+                    self.embedding_output_dir,
+                    write_to_filename=self.write_to_filename,
+                    output_type="parquet",
+                )
```

Comment:
I think we might want to move this up to line `207`, just to ensure that 

` embedding_ddf = self.create_embeddings(dataset.df, self.input_column)` is added to the profile code trace.

Reply:
Moved it in, but for future note, we tried profiling inside and out, and it doesn't matter where as long as an action / execution is happening within the `with profile(..)` context